### PR TITLE
PTX-12868 Check pure-telemetry-certs for owner reference before deleting StorageCluster

### PR DESCRIPTION
**Pull request by nikolaypopov**
Check pure-telemetry-certs for owner reference before deleting StorageCluster

Signed-off-by: nikolaypopov <nikolay.popov86@gmail.com>

### CODE REVIEW

File: pkg/util/test/util.go [hunk 1]
```diff
@@ -413,11 +413,45 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
...
+		}
 	}
 
 	return operatorops.Instance().DeleteStorageCluster(cluster.Name, cluster.Namespace)
 }
 
+func checkTelemetrySecretGotOwnerReference(cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
+	t := func() (interface{}, bool, error) {
+		secret, err := coreops.Instance().GetSecret("pure-telemetry-certs", cluster.Namespace)
+		if err != nil {
```
Comment (pure-jliao) [inline pkg/util/test/util.go:430]: as we check when telemetry enabled so the secret must exist. so we retry for all errors. if not found then there's an issue.

File: pkg/util/test/util.go [hunk 2]
```diff
@@ -413,11 +413,45 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
...
+		defaultCheckReferenceInterval := 1 * time.Second
+		if err = checkTelemetrySecretGotOwnerReference(cluster, defaultCheckReferenceTimeout, defaultCheckReferenceInterval); err != nil {
+			return err
+		}
 	}
 
 	return operatorops.Instance().DeleteStorageCluster(cluster.Name, cluster.Namespace)
 }
 
+func checkTelemetrySecretGotOwnerReference(cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
```
Comment (pure-jliao) [inline pkg/util/test/util.go:427]: if telemetry disabled, return nil directly, only check ref when telemetry enabled.

File: pkg/util/test/util.go [hunk 3]
```diff
@@ -413,11 +413,45 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
...
+	t := func() (interface{}, bool, error) {
+		secret, err := coreops.Instance().GetSecret("pure-telemetry-certs", cluster.Namespace)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return nil, false, nil
+			}
+		}
+		logrus.Debugf("Found secret %s", secret.Name)
+
+		if len(secret.OwnerReferences) != 0 {
```
Comment (pure-jliao) [inline pkg/util/test/util.go:437]:
```suggestion
		if len(secret.OwnerReferences) != 0 {
		ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
		for _, reference := range secret.OwnerReferences {
		    if reference.UID == ownerRef.UID {
		    }
		
```

**Comment by pure-jliao** [comment]
btw can we fix this at the same PR? https://github.com/libopenstorage/operator/blob/master/pkg/util/test/util.go#L3241
see here for reference https://github.com/libopenstorage/operator/blob/px-rel-1.10.0/pkg/util/test/util.go#L1959
want to add more coverage.

File: pkg/util/test/util.go (hunk 1 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:431]: Added a check for other errors

File: pkg/util/test/util.go (hunk 2 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:428]: Won't do this as secret might exist from previously enabled telemetry

File: pkg/util/test/util.go (hunk 3 above)
Comment (nikolaypopov) [reply pkg/util/test/util.go:439]: Done

**Review by pure-jliao — APPROVED** [review]
LGTM!

**Pull request merged**